ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/template2/Template2.kt

```diff
@@ -0,0 +1,160 @@
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.core.graphics.drawable.toBitmap
+import com.revenuecat.purchases.Package
+import com.revenuecat.purchases.ui.revenuecatui.PaywallViewModel
+import com.revenuecat.purchases.ui.revenuecatui.PaywallViewState
+import com.revenuecat.purchases.ui.revenuecatui.R
+import com.revenuecat.purchases.ui.revenuecatui.UIConstant
+import com.revenuecat.purchases.ui.revenuecatui.accent1Color
+import com.revenuecat.purchases.ui.revenuecatui.accent2Color
+import com.revenuecat.purchases.ui.revenuecatui.backgroundColor
+import com.revenuecat.purchases.ui.revenuecatui.callToActionBackgroundColor
+import com.revenuecat.purchases.ui.revenuecatui.callToActionForegroundColor
+import com.revenuecat.purchases.ui.revenuecatui.getActivity
+import com.revenuecat.purchases.ui.revenuecatui.getColors
+import com.revenuecat.purchases.ui.revenuecatui.localizedConfig
+import com.revenuecat.purchases.ui.revenuecatui.text1Color
+
+@Composable
+internal fun Template2(state: PaywallViewState.Template2, viewModel: PaywallViewModel) {
+    Column(
+        modifier = Modifier.fillMaxSize(),
+    ) {
+        Spacer(modifier = Modifier.weight(1f))
+        Template2MainContent(state, viewModel)
+        Spacer(modifier = Modifier.weight(1f))
+        PurchaseButton(state, viewModel)
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(
+                    start = UIConstant.defaultHorizontalPadding,
+                    end = UIConstant.defaultHorizontalPadding,
+                    bottom = UIConstant.defaultButtonVerticalSpacing,
+                ),
+            horizontalArrangement = Arrangement.Center,
+        ) {
+            RestorePurchasesButton(viewModel)
+        }
+    }
+}
+
+@Composable
+private fun Template2MainContent(state: PaywallViewState.Template2, viewModel: PaywallViewModel) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .verticalScroll(rememberScrollState())
+            .padding(horizontal = UIConstant.defaultHorizontalPadding),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.spacedBy(UIConstant.defaultButtonVerticalSpacing, Alignment.CenterVertically),
+    ) {
+        // TODO-PAYWALLS: Replace with correct image
+        val drawable = LocalContext.current.packageManager.getApplicationIcon(LocalContext.current.packageName)
+        Image(drawable.toBitmap(config = Bitmap.Config.ARGB_8888).asImageBitmap(), contentDescription = "")
```

Comment:
For now, I'm just using the App Icon. Once we implement a mechanism to actually use images from the dashboard we can use it.